File: codes/day-2/payrollapp/src/models/Hr.java
package models;
public class Hr extends Employee {
    private double gratuityPayment;

    public Hr(int id) {
        super(id);
    }

    public Hr(int id, String name, double basicPayment, double daPayment, double hraPayment, double gratuityPayment) {
        super(id, name, basicPayment, daPayment, hraPayment);
        this.gratuityPayment = gratuityPayment;
    }

    public double getGratuityPayment() {
        return gratuityPayment;
    }

    public void setGratuityPayment(double gratuityPayment) {
        this.gratuityPayment = gratuityPayment;
    }

    @Override
    public void calculateSalary() {
        super.calculateSalary();
        setTotalPayment(gratuityPayment + getTotalPayment());
    }
}

File: codes/day-12/payroll-app/src/models/hr.js
import { Employee } from "./employee.js";

export class Hr extends Employee {
  constructor(id, name, basic, da, hra, gratuity) {
    super(id, name, basic, da, hra);
    this.gratuityPayment = gratuity;
  }
  calculateSalary() {
    super.calculateSalary();
    this.totalPayment += this.gratuityPayment;
  }
}
